modules/005-external-module-manager/hooks/handle_sources.go

```diff
@@ -258,7 +300,38 @@ func fetchModuleVersion(logger *logrus.Entry, dc dependency.Container, moduleSou
 	return "v" + moduleMetadata.Version.String(), nil
 }
 
-func fetchAndCopyModuleVersion(dc dependency.Container, externalModulesDir string, moduleSource v1alpha1.ExternalModuleSource, moduleName, moduleVersion string, registryOptions []cr.Option) error {
+func fetchModuleWeight(moduleVersionPath string) int {
+	moduleDefFile := path.Join(moduleVersionPath, module_manager.ModuleDefinitionFileName)
+
+	if _, err := os.Stat(moduleDefFile); err != nil {
+		return defaultExternalModuleWeight
+	}
+
+	var def module_manager.ModuleDefinition
+
+	f, err := os.Open(moduleDefFile)
+	if err != nil {
+		return defaultExternalModuleWeight
+	}
+	defer f.Close()
+
+	err = yaml.NewDecoder(f).Decode(&def)
+	if err != nil {
+		return defaultExternalModuleWeight
+	}
+
+	if def.Weight > 0 && def.Weight < 100 {
+		def.Weight = defaultExternalModuleWeight + def.Weight
+	}
+
+	if def.Weight < defaultExternalModuleWeight || def.Weight >= 1000 {
+		def.Weight = defaultExternalModuleWeight
+	}
```

Comment:
Do we have validation of module.yaml while downloading the module?
Can we validate the module.yaml against a schema?

Probably we need to introduce versioning to make schema changes like in shell operator.

Reply:
create an issue: https://github.com/flant/addon-operator/issues/402